### ISSUE TITLE
EpiChrome 2.3.1 can't create an SSB

### ISSUE THREAD

**Issue by adamwunn** [dmarmor/epichrome#223]
I am having a problem with EpiChrome on one of my Macs. It works on another Mac.

I am able to step through the wizard, but when I try to actually save/create the SSB, I get the following error:  "Creation failed: Unable to set app executable permissions."

In the log I see:

chmod: /Users/username/Desktop/Google Docs.app.1796/Contents.29054/MacOS/Epichrome: No such file or directory
*[9922]Epichrome|build.sh(183)/updateapp(243): Unable to set app executable permissions.
![9922]Epichrome|build.sh(184)/myabort(69)/abortsilent(861)/abort(834): Aborting: Unable to set app executable permissions.

The "1796" and "29054" on the contents might be part of the issue. I didn't add the 1796 to my app name and I have never seen anything after "Contents" in bundles before. If the app is truly using the command that is shown in the log, I can see why it is failing. Those names are not real. That is unless the app is using these temporary names for a purpose and I am just not understanding. Either way, I can't make an SSB. Any ideas? (2.24 still works fine for me.)

**Comment by dmarmor** [comment]
Hmm. I'm not sure why you're not able to set the executable permissions here. It appears you're creating the app inside your own home directory, so you should be able to do so. Is there any difference in your security settings or user level between the two Macs?

(Don't worry about the "1796" and "29054" -- those numbers are added to the end of temp directories as the app is being built so that they don't interfere with the old directory during an update, and the old version can be restored on error.)

**Comment by adamwunn** [comment]
Hello,

Thanks for your response. The way I “solved” the problem with that one SSB I mentioned before was to create it on another computer and copy it back to this computer. Now, all of a sudden ALL of my SSB’s created with EpiChrome won’t open or update. They generate logs like this below. Any idea what what is going on? This is driving me nuts. I am very sad, because several of SSBs have a ton of work done in them with security authorizations for a ton of customer remote login profiles. I would hate to have to try this again with an inferior SSB. Before this EpiChrome has been wonderful and the BEST SSB I’ve ever used. I can’t imagine what is wrong with this one Mac. I run a tight ship and I keep it clean and stable. I am still running 10.12.6, because I am a stability freak, could the old OS have anything to do with it? The Mac I created the SSB successfully on recently was a 10.15 Catalina Mac (boy I hate that OS).

2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: openssl:Error: 'AES-128-CBC' is an invalid command.
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: Standard commands
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: asn1parse      ca             ciphers        crl            crl2pkcs7      
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: dgst           dh             dhparam        dsa            dsaparam       
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: ec             ecparam        enc            engine         errstr         
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: gendh          gendsa         genrsa         nseq           ocsp           
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: passwd         pkcs12         pkcs7          pkcs8          prime          
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: rand           req            rsa            rsautl         s_client       
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: s_server       s_time         sess_id        smime          speed          
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: spkac          verify         version        x509           
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: Message Digest commands (see the `dgst' command for more details)
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: md2            md4            md5            mdc2           rmd160         
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: sha            sha1           
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: Cipher commands (see the `enc' command for more details)
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: aes-128-cbc    aes-128-ecb    aes-192-cbc    aes-192-ecb    aes-256-cbc    
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: aes-256-ecb    base64         bf             bf-cbc         bf-cfb         
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: bf-ecb         bf-ofb         cast           cast-cbc       cast5-cbc      
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: cast5-cfb      cast5-ecb      cast5-ofb      des            des-cbc        
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: des-cfb        des-ecb        des-ede        des-ede-cbc    des-ede-cfb    
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: des-ede-ofb    des-ede3       des-ede3-cbc   des-ede3-cfb   des-ede3-ofb   
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: des-ofb        des3           desx           rc2            rc2-40-cbc     
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: rc2-64-cbc     rc2-cbc        rc2-cfb        rc2-ecb        rc2-ofb        
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: rc4            rc4-40         seed           seed-cbc       seed-cfb       
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(239) /{openssl}: seed-ecb       seed-ofb       
2[867]GoogleAccoun(404)/updatessb(135) /updateapp(243) /{chmod}: chmod: /Applications/Google Accounts.app/Contents.28602/MacOS/Epichrome: No such file or directory
*[867]GoogleAccoun(404)/updatessb(135) /updateapp(243): Unable to set app executable permissions.
*[867]GoogleAccoun(529)/alert(1440)/dialog(1391): Showing alert 'Warning' with text 'Unable to update app to new version of Google Chrome. (Error filtering Info.plist file. (ascii)) It may or may not run properly.'
![867]GoogleAccoun(536)/abort(834): Aborting:

**Comment by dmarmor** [comment]
This actually looks to be possibly the same problem. If it is what I think it is, then I'm guessing you've *never* been able to successfully create a new app on this computer. I suspect the problem is that 10.12 uses an earlier, incompatible version of `openssl` which I was using to create the main app executable.

The good news is that, as of 2.3.2, I'm no longer using `openssl` at all. I still can't guarantee it will be compatible with 10.12 as I don't have that system installed anywhere to test, but I'd be grateful if you could install 2.3.2 and try creating a new app with it and let me know how it goes.

Thanks!

**Comment by adamwunn** [comment]
Thanks for your response. I was able to create all of my SSBs with version 2.2 and one with 2.2.4. I think you are right about the openssl, because I was able to create the “new” SSB on the second machine with 2.3 and it was using Mac OS 10.15.3. Now that I downloaded 2.3.2, the process is working again as you suspected as well! What a relief. I was able to install the new version and then update the existing SSBs. No more errors! Yahhhhhhhhh! Thank you for your diligent and timely help. It is greatly appreciated. Epichrome is a critical staple of my world.
